src/effects/native/autopaneffect.cpp

```diff
@@ -99,27 +99,30 @@ void AutoPanEffect::processChannel(const ChannelHandle& handle, AutoPanGroupStat
     AutoPanGroupState& gs = *pGroupState;
     double width = m_pWidthParameter->value();
     double period = m_pPeriodParameter->value();
-    double smoothing = 0.5-m_pSmoothingParameter->value();
+    double smoothing = 0.5 - m_pSmoothingParameter->value();
 
     if (groupFeatures.has_beat_length_sec) {
         // period is a number of beats
         double beats = std::max(roundToFraction(period, 2), 0.25);
-        // NOTE: Assuming engine is working in stereo.
-        period = beats * groupFeatures.beat_length_sec * sampleRate * 2;
+        period = beats * groupFeatures.beat_length_sec * sampleRate;
```

Comment:
I hear that this cycles the sound around the stereo field in the correct amount of time, but I do not understand why it was wrong to multiply by the channel count.

Reply:
The stereo assumption is already handled in the loop below  i += 2